tests/rptest/tests/nodes_decommissioning_test.py

```diff
@@ -112,10 +112,9 @@ def requested_status():
         wait_until(requested_status, timeout_sec=timeout_sec, backoff_sec=1)
 
     def _set_recovery_rate(self, rate):
-        # using rpk tool here not to check the versions of a config on each node
-        # as throttling recovery may prevent configuration to be delivered to all nodes
-        rpk = RpkTool(self.redpanda)
-        rpk.cluster_config_set("raft_learner_recovery_rate", str(rate))
+
+        self.redpanda.set_cluster_config({"raft_learner_recovery_rate": rate},
```

Comment:
If the test knows which are up, can `skip_readiness_check` explicitly list of the nodes that aren't expected to catch up?  It seems like a "big hammer" to make the function skip this entirely.

Reply:
It feels very weird that the tests request a change but doesn't wait until it's applied. The question is why does it need to request it in the first place if the test doesn't control when it's applied? It looks like we introduce a race, make the test less deterministic and potentially don't test what we want to test.

Reply:
the test doesn't have to wait for all of the nodes to apply configuration as the partitions are evenly distributed across the cluster and throttling learners recovery even on a single node is enough to prevent nodes decommissioning from finishing.

Reply:
Then we probably should use `any` to make sure it's applied at least on one node - https://github.com/redpanda-data/redpanda/pull/8497#discussion_r1091087175

Reply:
We do not need to wait for a single node as the `patch_cluster_config` returns after config change is replicated and applied on the leader so it is enough to relay on a request being successful